### ISSUE TITLE
`SemanticsService.announce` never completes when Accessibility service is disabled / when device doesn't have accessibility options

### ISSUE THREAD

**Issue by PieterAelse** [flutter/flutter#94100]
## Steps to Reproduce

Use this code:
```dart
print("PRE-ANNOUNCEMENT: ${DateTime.now()}");
await SemanticsService.announce("This is a semantics announcement", Directionality.of(context));
print("POST-ANNOUNCEMENT: ${DateTime.now()}");
```

With e.g. TalkBack (on Android) on and off. I noticed that when TalkBack is turned off the second print is **not printed anymore**, meaning the `.announce` never completes when the service is off? That doesn't seem right. Now I need to wrap all my announcements in `unawaited(...)`.

**Expected results:** 
Future always completes (when the message is delivered to the native side)

**Actual results:** 
Future **does not complete** when e.g. TalkBack is turned off.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: ElevatedButton(
            child: const Text('Tap here to do a semantics announcement'),
            onPressed: () async {
              print("PRE-ANNOUNCEMENT: ${DateTime.now()}");
              await SemanticsService.announce("This is a semantics announcement", Directionality.of(context));
              print("POST-ANNOUNCEMENT: ${DateTime.now()}");
            },
          ),
        ),
      ),
    );
  }
}

```

</details>

Used Flutter version 2.5.2.

<details>

<summary>Flutter doctor output</summary>

```
mac[redacted]:Code pieter$ flutter doctor -v
[✓] Flutter (Channel unknown, 2.5.2, on macOS 11.6 20G165 darwin-x64, locale en-US)
    • Flutter version 2.5.2 at [redacted]
    • Upstream repository unknown
    • Framework revision 3595343e20 (8 weeks ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/pieter/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/pieter/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2021.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.0.4
    • Dart plugin version 212.5632

[✓] Connected device (3 available)
    • Pixel 3a (mobile) • [redacted] • android-arm64  • Android 12 (API 31)
    • macOS (desktop)   • macos      • darwin-x64     • macOS 11.6 20G165 darwin-x64
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 96.0.4664.55
```
</details>

**Comment by danagbemava-nc** [comment]
Hi @PieterAelse, thanks for filing the issue. 

Please provide a complete minimal reproducible code sample and the output of `flutter doctor -v`. 

Thank you

**Comment by PieterAelse** [comment]
Hi @danagbemava-nc  I've added a minimal reproducible code sample to the "Code sample" section in the OP and added "Flutter doctor output" too.

The code sample can be run in Dartpad, you'll see that the second print is not visible in the console (it will only be when running on an actual device and accessibility service, e.g. TalkBack, is enabled).

**Comment by danagbemava-nc** [comment]
Reproducible on `stable`, `master`, `beta`.

<details>
<summary>sample code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Builder(
        builder: (BuildContext context) {
          return Scaffold(
              body: Center(
            child: ElevatedButton(
              child: const Text('Tap here to do a semantics announcement'),
              onPressed: () async {
                print("PRE-ANNOUNCEMENT: ${DateTime.now()}");
                await SemanticsService.announce(
                    "This is a semantics announcement",
                    Directionality.of(context));
                print("POST-ANNOUNCEMENT: ${DateTime.now()}");
              },
            ),
          ));
        },
      ),
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-arm, locale en-GB)
    • Flutter version 2.5.3 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (6 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.0, Build version 13A233
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (5 available)
    • M2007J20CG (mobile)    • 5dd3be00                             • android-arm64  • Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E            • ios            • iOS 15.0.2 19A404
    • iPhone 13 (mobile)     • 7515F01A-C1B6-456E-8559-8D2383D3719D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.6 20G165 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.55

• No issues found!
```

```console
[✓] Flutter (Channel master, 2.6.0-12.0.pre.826, on macOS 11.6 20G165 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.826 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c1455a23a (11 hours ago), 2021-11-23 21:13:05 -0500
    • Engine revision 539239e256
    • Dart version 2.16.0 (build 2.16.0-42.0.dev)
    • DevTools version 2.8.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (5 available)
    • M2007J20CG (mobile)    • 5dd3be00                             • android-arm64  • Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E            • ios            • iOS 15.0.2 19A404
    • iPhone 13 (mobile)     • 7515F01A-C1B6-456E-8559-8D2383D3719D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.6 20G165 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.55

• No issues found!
```

```console
[✓] Flutter (Channel beta, 2.8.0-3.1.pre, on macOS 11.6 20G165 darwin-arm, locale en-GB)
    • Flutter version 2.8.0-3.1.pre at /Users/nexus/dev/sdks/flutter_beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3dab555040 (12 days ago), 2021-11-12 11:50:00 -0800
    • Engine revision 09f1520e8b
    • Dart version 2.15.0 (build 2.15.0-268.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (5 available)
    • M2007J20CG (mobile)    • 5dd3be00                             • android-arm64  • Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E            • ios            • iOS 15.0.2 19A404
    • iPhone 13 (mobile)     • 7515F01A-C1B6-456E-8559-8D2383D3719D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 11.6 20G165 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.55

• No issues found!
```
</details>

**Comment by goderbauer** [comment]
This does sound like a bug. The future should always complete. /cc @chunhtai

**Comment by chunhtai** [comment]
I looked at the code, it future should resolve unless the method call crashed the engine side, I will need to look into it
